mne/io/curry/tests/test_curry.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+#
+# Authors: Dirk Gütlin <dirk.guetlin@stud.sbg.ac.at>
+#
+#
+# License: BSD (3-clause)
+
+import os.path as op
+import numpy as np
+
+import pytest
+
+from numpy.testing import assert_allclose, assert_array_equal
+from mne.annotations import events_from_annotations
+from mne.datasets import testing
+from mne.event import find_events
+from mne.io.constants import FIFF
+from mne.io.edf import read_raw_bdf
+from mne.io.bti import read_raw_bti
+from mne.io.curry import read_raw_curry
+from mne.io.curry.curry import _check_missing_files, _read_events_curry
+from mne.utils import check_version
+
+
+data_dir = testing.data_path(download=False)
+curry_dir = op.join(data_dir, "curry")
+
+bdf_file = op.join(data_dir, 'BDF', 'test_bdf_stim_channel.bdf')
+
+bti_rfDC_file = op.join(data_dir, 'BTi', 'erm_HFH', 'c,rfDC')
+
+curry7_rfDC_file = op.join(curry_dir, "c,rfDC Curry 7.dat")
+curry8_rfDC_file = op.join(curry_dir, "c,rfDC Curry 8.cdt")
+
+curry7_bdf_file = op.join(curry_dir, "test_bdf_stim_channel Curry 7.dat")
+curry7_bdf_ascii_file = op.join(curry_dir,
+                                "test_bdf_stim_channel Curry 7 ASCII.dat")
+
+curry8_bdf_file = op.join(curry_dir, "test_bdf_stim_channel Curry 8.cdt")
+curry8_bdf_ascii_file = op.join(curry_dir,
+                                "test_bdf_stim_channel Curry 8 ASCII.cdt")
+
+test_sfreq_0 = op.join(curry_dir, "test_sfreq_0.dat")
+
+if not check_version("numpy", "1.16.0"):
+    do_warn = 'ignore:.*take longer for ASCII.*:'
+else:
+    do_warn = ''
+
+
+@pytest.fixture(scope='session')
+def bdf_curry_ref():
+    """Return a view of the reference bdf used to create the curry files."""
+    raw = read_raw_bdf(bdf_file, preload=True).drop_channels(['Status'])
+    return raw
+
+
+@pytest.mark.filterwarnings(do_warn)
+@testing.requires_testing_data
+@pytest.mark.parametrize('fname,tol', [
+    pytest.param(curry7_bdf_file, 1e-7, id='curry 7'),
+    pytest.param(curry8_bdf_file, 1e-7, id='curry 8'),
+    pytest.param(curry7_bdf_ascii_file, 1e-4, id='curry 7 ascii'),
+    pytest.param(curry8_bdf_ascii_file, 1e-4, id='curry 8 ascii'),
+    pytest.param(test_sfreq_0, 1e-7, id='test sfreq = 0'),
```

Comment:
There's something I don't understand, we have a sfreq=0 and everything works?

Reply:
Yes, sfreq reconstructed from the time steps. You pointed out to cover this here:
https://github.com/mne-tools/mne-python/pull/6272/files#r290267509

But since we're speaking of that; If you only want to read curry annotations (without reading entire curry files), with `read_annotations` or `_read_annotations_curry`, it does not do this trick. It only tries to read `sfreq` from the according file (similar to the brainstorm annotations reader). But of course you can pass `sfreq` manually for these funcs. I'm not sure whether i should keep this strictly in line with the other annotations, or add the alternative to create it from time steps.
What do you think?

Reply:
I was expecting something like this:

```py
def test_0_sfreq():
    with pytest.raises(RuntimeError, match='sfreq * 0'):
        read_raw_curry(test_sfreq_0)
```

Reply:
+ a big crash!

Reply:
oh, sorry. Now I understand. `sreq` is 0 but `time_step` is not. The names confused me. 

I would mock the file and do a unit test for that, rather than using an integration test. 

```py
@pytest.fixture(params=['sfreq', 'time_step', pytest.param('no sfreq', marks=pytest.mark.raises)])
def sfreq_testig_data(request):
	if request == 'sfreq':
		return curryX_bdf_file
    else:
        # create tmpdir
        # copy curryX_bdf_file
        # modify the file
        return path

def test_data(sfreq_testing_data):
    assert read_raw_curry(sfreq_testing_data).info['sfreq'] = 42
```

Reply:
>plus a big crash!

Yes, sure. If `sfreq = 0` and `time_step == inf`, and for some reason sfreq would still be == 0 after the step mentioned in https://github.com/mne-tools/mne-python/pull/6272/files#r290267509  ,
this would be catched and covered in `create_info` and return:
```python
        if sfreq <= 0:
>           raise ValueError('sfreq must be positive')
E           ValueError: sfreq must be positive
```
I thought adding one single file would be sufficient, since this step is the same for all data and the idea was only to cover these 2 additional lines.

So, was your idea to change this in order to be able to test multiple files instead of only one?